jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/ReopenCaseCommand.java

```diff
@@ -79,20 +79,22 @@ public Void execute(Context context) {
         
         caseEventSupport.fireBeforeCaseReopened(caseId, caseFile, deploymentId, caseDefinitionId, data);
         
-        logger.debug("Updating case file in working memory");
+        logger.debug("Removing case file from working memory to allow refiring of rules");
         FactHandle factHandle = ksession.getFactHandle(caseFile);
+        ksession.delete(factHandle);
         ((CaseFileInstanceImpl)caseFile).setCaseReopenDate(new Date());
         if (data != null && !data.isEmpty()) {
             caseFile.addAll(data);
         }
-        ksession.update(factHandle, caseFile);
+        
         
         logger.debug("Starting process instance for case {} and case definition {}", caseId, caseDefinitionId);
         CorrelationKey correlationKey = correlationKeyFactory.newCorrelationKey(caseId);
         Map<String, Object> params = new HashMap<>();
         // set case id to allow it to use CaseContext when creating runtime engine
         params.put(EnvironmentName.CASE_ID, caseId);
         long processInstanceId = processService.startProcess(deploymentId, caseDefinitionId, correlationKey, params);
+        ksession.insert(caseFile);
```

Comment:
Shouldn't _caseFile_ be inserted  before starting the process?

Reply:
no, that's the thing if it is then it will fire rules before any of the listeners is triggered

Reply:
yes, I've tested that approach, but some assertions are failing, because without the caseFile_ prefix, a new variable is created, so expecting 3 instead of 4:
caseFile_name vs name (new one).